### ISSUE TITLE
Importing controlsfx into Eclipse / Buildship fails

### ISSUE THREAD

**Issue by JonathanGiles** [controlsfx/controlsfx#937]
**[Original report](https://bitbucket.org/controlsfx/controlsfx/issue/746) by Dean Schulze (Bitbucket: [dwschulze](https://bitbucket.org/dwschulze), GitHub: [dwschulze](https://github.com/dwschulze)).**

----------------------------------------

I installed the Buildship Gradle plugin into Eclipse Neon.  When I import the controlsfx project (using the wrapper) it fails with this error:

java.lang.IllegalStateException: Workspace already contains a project with name controlsfx.

I've deleted the workspace and imported again with the same error.

When this happens all I see in the Project Explorer for the controlsfx project is a gradle/ directory with the wrapper, but no source directories or files.

This may be an Eclipse of Buildship bug.  I'm open to using another IDE if I have to.  

I tried the Egradle plugin for Eclipse and it also failed to import controlsfx, but it gave a different error.

**Comment by wirew0rm** [comment]
Maybe it would be possible to add instructions on how to import controlsfx into an eclipse (or any other IDE) workspace properly? I tried importing as a Gradle project and then renaming the project. This kind of worked, but running the samples then fails with 'project not found controlsfx-controlsfx' and the project settings are strange because some of them are checked into the repo but get overwritten by Gradle and/or Eclipse.

Issue #173 has instructions for IntelliJ, but I'm not sure if they are still up to date, but if they are, they should definitely be included or linked in the README.md. I tried using a similar approach in Eclipse, but it yields the same result as my importing and renaming approach.

While it would be nice to fix the underlying issues with the project structure, I think just documenting the workarounds needed to get a working eclipse workspace would significantly lower the barrier to contribute to controlsfx.

For what it's worth #1057 seems to be a duplicate of this bug and should probably be closed in favor of this one.

**Comment by abhinayagarwal** [comment]
Hi @wirew0rm,

Thank you for testing out controlsfx on Eclipse.

To help you better, can you add the Eclipse version used for importing ControlsFX and does Eclipse include the Gradle Buildship plugin?

**Comment by wirew0rm** [comment]
Eclipse Version: 2019-03 (4.11.0)
Eclipse Buildship	Buildship, Eclipse Plug-ins for Gradle	3.0.1.v20181217-1554	org.eclipse.buildship

I just reproduced everything in a clean workspace and on a clean checkout of the 9.0.0 branch:
Import -> Existing Gradle Project -> $workspace/controlsfx
I got 2 errors:
  * Missing Nature: org.eclipse.jdt.groovy.core.groovyNature (Popup)
    * When I try to install it from the Marketplace I get dependency Errors
  * Workspace already contains a project with name controlsfx. (Problems Tab)

I tried to prefix the `controlsfx` and `fxsampler` subdirs with `controlsfx-` and changed the references in  
- .project
- controlsfx-controlsfx/.project
- controlsfx-fxsampler/.project
- controlsfx-samples/.classpath
- controlsfx-samples/build.gradle
- settings.gradle

The actual "Workspace already contains..." error seems to be caused by using the same name in the .project file of the root project and the actual controlsfx library.
Now I can import the project and start the sample from the "Gradle Task" tab in eclipse.
The dependency resolution inside Eclipse unfortunately still does not work, because the eclipse settings files add the Dependencies determined by Gradle to the Classpath instead of the Modulepath and also do not correctly set the module access rules.
 [Here](https://github.com/wirew0rm/controlsfx/tree/eclipse-importable) are the changes i applied so far.

Also just loading the project makes big changes to the source controlled eclipse settings, so if they are not absolutely necessary, maybe this configuration should be moved into build.gradle with the eclipse gradle plugin.

**Comment by abhinayagarwal** [comment]
Hi @wirew0rm,

Thank you for taking time and jotting down all the details. I think its high time we remove all the eclipse related files from the repository. Eclipse can depend on Gradle for everything.

I followed the following steps and I was able to import the project(s) successfully. The dependency resolution, as you said, needs some work. Can you try the following:

1. Delete ControlsFX from Eclipse workspace.
2. Close Eclipse
3. Delete `.project`, `.classpath` and `.settings` from ControlsFX
4. Create `gradle.properties` and add `org.gradle.java.home=<path-to-jdk-9+>`
5. Re-open Eclipse and go to File -> Import -> Gradle -> Existing Gradle Project
6. Select ControlsFX and click on Finish

All projects should be imported successfully.

**Comment by wirew0rm** [comment]
Yes, I can confirm, changing the directory structure is not necessary, just removing the eclipse files is enough (Also gets rid of the missing nature popup).
The `gradle.properties` file did not make any difference for me (openjre11 is the default jre anyway).
For the Module Dependencies the only thing I could find was this ~2 years old stackoverflow [How to to tell Eclipse/Buildship/Gradle to add dependencies to the Modulepath](https://stackoverflow.com/a/47820771), I didn't manage to get it working and I'm not sure if this is still up to date.

**Comment by abhinayagarwal** [comment]
@wirew0rm What about the master branch? Does it still need Eclipse related files? I think Buildship should be handle it for JDK 8 as well.

It would be great if we can get rid of all Eclipse related files in both the branches.

**Comment by wirew0rm** [comment]
On the master branch I have to check "Override workspace settings" and choose a more recent gradle version (in my case 5.5-rc-3). Else I get an 'Could not determine Java version from 11.0.3' which seems to originate from gradle using some deprecated API to check the version. On the 9.0.0 branch this is not necessary, I suspect that it contains a newer gradle wrapper.

I also have to delete the Eclipse files (Probably the .projects are sufficient but I deleted everything eclipse related) to avoid the "project already exists" error but then I can just import and use "Run as java application" on `ControlsFXSampler.java` and everything works as it should, all the imports are resolved correctly, I can use F3 to jump between the different subprojects and autocomplete and documentation is also working.

The only errors that are still shown are "discouraged access" type errors, I don't know if the project files in the repo flipped the eclipse switch which makes these warnings instead of errors, this would anyhow be nice if we could enable this from gradle.

**Comment by scasadei** [comment]
Hello @abhinayagarwal,
I tried to install controlsFX (master branch) on Eclipse (1912 version) by following your instructions of June 19 2019, it failed with this message:

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.0.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 1s

Thanks for any help,
Stefano
